### ISSUE TITLE
deadlock in WPF

### ISSUE THREAD

**Issue by ken-wangshoudong** [max-ieremenko/ServiceModel.Grpc#50]
.net 5 wpf client
------------------------------------------------
        private async Task<int> Test()
        {
            return await proxy.Sum(10, 5).ConfigureAwait(false); 
        }
        private  void Button_Click(object sender, RoutedEventArgs e)
        {          
            var ttt = Test().Result; // deadlock.
            MessageBox.Show(ttt.ToString());
        }

.net 5 grpc service
   public  Task<int> Sum(int a, int b)
        {
            return Task.FromResult(a + b);
        }

how to fix it?.  thanks

**Comment by max-ieremenko** [comment]
Thank you for the bug report.

The bug is in the ServiceModel.Grpc client call implementation: attempt to marshal the continuation back to the original captured SynchronizationContext. It leads to deadlock when you make a call from WPF main thread.

Deadlock also can be reproduced by the following code:
``` c#
    private async Task<int> Test()
    {
        await Task.Delay(1000);
        return 1; 
    }
    private  void Button_Click(object sender, RoutedEventArgs e)
    {          
        var ttt = Test().Result; // deadlock
        MessageBox.Show(ttt.ToString());
    }
```
Fix for ServiceModel.Grpc comes in 1.3.2 version.

How to fix in your code:

**variant 1**

you code expects "blocking unary call", so change contract accordingly:
``` c#
    // grpc service
    public int Sum(int a, int b) => a + b;

    // WPF
    private  void Button_Click(object sender, RoutedEventArgs e)
    {          
        var ttt = proxy.Sum(10, 5); // blocking unary call
        MessageBox.Show(ttt.ToString());
    }
```

**variant 2**

make button click handler async:
``` c#
    private async Task<int> Test()
    {
        return await proxy.Sum(10, 5).ConfigureAwait(false); 
    }
    private async void Button_Click(object sender, RoutedEventArgs e)
    {          
        var ttt = await Test();

        // identical call 
        //// var ttt = await Test().ConfigureAwait(true);

        MessageBox.Show(ttt.ToString());
    }
```
details and explanation are [here](https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming).

**Comment by max-ieremenko** [comment]
The fix for ServiceModel.Grpc is released in 1.3.2.

**Comment by ken-wangshoudong** [comment]
> The fix for ServiceModel.Grpc is released in 1.3.2.
just now ,i already succeeded run the code . 

thank you very much.
